src/python/pants/backend/experimental/jvm/target_types.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import logging
+
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    Dependencies,
+    Sources,
+    StringSequenceField,
+    Target,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class JavaSources(Sources):
+    expected_file_extensions = (".java",)
+
+
+class JavaTestsSources(JavaSources):
+    default = ("*Test.java",)
+
+
+class JavaTests(Target):
+    alias = "java_tests"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        Dependencies,
+        JavaTestsSources,
+    )
+    help = "Java tests"
```

Comment:
@Eric-Arellano:  Should the target be called `junit_tests`? I know there was some thought in Python around using `pytest_tests` instead of `python_tests`?

Reply:
I think probably. I wish that we had named it `pytest_tests` and had proposed changing it, but people requested we don't to avoid churn.

We named it `shunit2_tests`, rather than `shell_tests`.

It in part comes down to if there are other test runners for Java that we might want to support. Also, it's possible to run Scala tests with Junit, right? If so, would we want to use the same target or distinct ones? Do they have the same metadata requirements or we want things like `scala_version` on the tests target?

> it's possible to run Scala tests with Junit, right? 

Even if possible, would we do this, or use Scalatest?

Reply:
Yeah, probably, test runners are so different, that naming the target for them makes the most sense.

Reply:
Done